### ISSUE TITLE
[All][Accessibility] [Lack of accessibility support for error messages with the new Input feature]

### ISSUE THREAD

**Issue by jwoo-msft** [microsoft/AdaptiveCards#4912]
# All platforms
- [x]  Android (#4958)
- [x] iOS (#4966)
- [x] UWP
- [x] .NET WPF
- [x] JS

# Host

Microsoft Teams

# Version of SDK

SDK version 2.1.0 (ACv1.3)

# Details
reference #4913 
When there are errors on the cards after pressing submit, failure to submit due to errors is not called to users. The user also not aware if there are multiple errors on the card.  
Proposed fix: Call out "There are errors within the card" 
Link to [video](https://msit.microsoftstream.com/video/af0ba1ff-0400-9fb2-d848-f1eb08a2e1e4)

**Comment by jwoo-msft** [comment]
@almedina-ms only for Android & 20.10 is TBD

**Comment by shiladityasaha** [comment]
@shalinijoshi19 @jwoo-msft this is a ring blocker on all platforms for us. Can it be picked up for other platforms too?

**Comment by shalinijoshi19** [comment]
Current status : P0's for Teams-impacting platforms needed for 20.10
[P0] - Android: @golddove working on changes.
[P0] - JS: @dclaux are you able to help out on this for JS? 
[P0] - iOS: @jwoo-msft is this on your radar? 
[P1] - UWP: @RebeccaAnne will need your help on this one and WPF to check out current state and changes that might be needed here 
[P1] - WPF: @RebeccaAnne

**Comment by RebeccaAnne** [comment]
Per discussion the other day, I believe we decided that moving the focus to the first error input and reading out that input's error would be our approach for this bug. @golddove can you confirm whether that solution is the official POR here? Is there another  issue tracking that other than this one?

**Comment by jwoo-msft** [comment]
@shalinijoshi19 yes, this is the issue I will be tackling on Today.

**Comment by jwoo-msft** [comment]
@shalinijoshi19  @RebeccaAnne @golddove 
There has been a concern that error message can not be controlled by Teams or host app.

**Comment by shalinijoshi19** [comment]
The plan of action here is to get into a debugging session tomorrow AM with the India team - @golddove @RebeccaAnne 
 are you able to get on a call tomorrow pls? Ideally we'd like to go over the precise scenario, and go over our proposal to make sure it will work.

**Comment by dclaux** [comment]
> Per discussion the other day, I believe we decided that moving the focus to the first error input and reading out that input's error would be our approach for this bug. 

That is already the way JS functions, and as @RebeccaAnne says was the plan per spec.

**Comment by golddove** [comment]
@RebeccaAnne

Correct, recent discussion resulted in the following resolutions:

1) When an Input receives focus, announce the input's label to screen readers. If the input has failed validation, append the error message to that announcement.

Sample accessible text for an invalidated username input: "Username, Usernames may only contain alphanumeric characters, Input"

2) On submit, if an error is present, focus moves to the first invalid input.

@dclaux does JS currently announce error messages in this way?

**Comment by shalinijoshi19** [comment]
Speclet referenced here https://github.com/microsoft/AdaptiveCards/pull/4929/ - The spec could do with some clarification btw on precisely the POA and moving down the rest of the options

**Comment by golddove** [comment]
Verified on JS. Works as expected.

**Comment by RebeccaAnne** [comment]
Confirmed this also works as expected in UWP

**Comment by shalinijoshi19** [comment]
Current status : P0's for Teams-impacting platforms needed for 20.10
[P0] - Android: In PR
[P0] - iOS: @jwoo-msft how are we tracking on this for iOS? 
Thanks

**Comment by jwoo-msft** [comment]
@shalinijoshi19 
it's almost done. I need to adjust 2-3 inputs before I can call it done.
